### PR TITLE
Add "limit=-1" support for the API and CLI and block negative values for limit query param

### LINKED ISSUE #3708: v2.3 Running the "st2 key list" command returns 100 entries vs all
In v2.3 when running the "st2 key list" command, the result returns 100 entries vs all.
The following was run in v2.1:
```
st2 key list -t xxxxxxx -j|grep "name"|wc -l
124
```

The following was run in v2.3:
```
st2 key list -t xxxxxxx -j|grep "name"|wc -l
100
```

### LINKED ISSUE #3735: Add '--last -1' to list all results from CLI
> Related to https://github.com/StackStorm/st2/issues/3736

Following the https://github.com/StackStorm/st2/issues/3708#issuecomment-328666287 consensus, there should be a way to to get all the results for any command in CLI via `--last -1`.

----

For reference, Mistral uses similar `--limit -1`
```
optional arguments:
  -h, --help            show this help message and exit
  --filter FILTERS      Filters. Can be repeated.
  --limit [LIMIT]       Maximum number of tasks to return in a single result.
                        limit is set to 100 by default. Use --limit -1 to
                        fetch the full result set.
```

Another example, `docker ps` CLI by default returns all possible results (via `--last -1`):

Name | default | Description
-- | -- | --
--last, -n | -1 | Show n last created containers (includes all states)

https://docs.docker.com/engine/reference/commandline/ps/#usage

### PR DESCRIPTION
**Pull request by humblearner**
Closes #3735, Closes #3708 Partially Addresses: #3736

**Before:**
```
$ st2 execution list -n -1
ERROR: 500 Server Error: Internal Server Error
MESSAGE: Internal Server Error for url: http://127.0.0.1:9101/v1/executions/?limit=-1&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null
```
```
$ st2 execution list -n -11
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| id                       | action.ref | context.user | status                 | start_timestamp         | end_timestamp          |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| 59c4ad9a909a5049b27360ab | core.local | stanley      | succeeded (0s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:42 UTC            | 06:28:42 UTC           |
| 59c4ad9c909a5049b27360ae | core.local | stanley      | succeeded (1s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:44 UTC            | 06:28:45 UTC           |
| 59c4ada3909a5049b27360b1 | core.local | stanley      | succeeded (0s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:51 UTC            | 06:28:51 UTC           |
| 59c4ada5909a5049b27360b4 | core.local | stanley      | succeeded (1s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:53 UTC            | 06:28:54 UTC           |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
+----------------------------------------------------------------------------------------------------------------------------------+
| Note: Only first -11 action executions are displayed. Use -n/--last flag for more results.                                       |
+----------------------------------------------------------------------------------------------------------------------------------+

```

**After:**
```
$ st2 execution list -n -1
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| id                       | action.ref | context.user | status                 | start_timestamp         | end_timestamp          |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| 59c4ad9a909a5049b27360ab | core.local | stanley      | succeeded (0s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:42 UTC            | 06:28:42 UTC           |
| 59c4ad9c909a5049b27360ae | core.local | stanley      | succeeded (1s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:44 UTC            | 06:28:45 UTC           |
| 59c4ada3909a5049b27360b1 | core.local | stanley      | succeeded (0s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:51 UTC            | 06:28:51 UTC           |
| 59c4ada5909a5049b27360b4 | core.local | stanley      | succeeded (1s elapsed) | Fri, 22 Sep 2017        | Fri, 22 Sep 2017       |
|                          |            |              |                        | 06:28:53 UTC            | 06:28:54 UTC           |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
```
```
$ st2 execution list -n -11
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-11" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/executions/?limit=-11&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null

$ curl -X GET -H  'Connection: keep-alive' -H  'Accept-Encoding: gzip, deflate' -H  'Accept: */*' -H  'User-Agent: python-requests/2.14.2' -H  'X-Auth-Token: 46a703f653f84629b5ac15cd145e69fc' 'http://127.0.0.1:9101/v1/executions/?limit=-1&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null' | grep -c "start_timestamp"

4
```
and block negative integers in limit query param:

```
$ curl -X GET -H  'Connection: keep-alive' -H  'Accept-Encoding: gzip, deflate' -H  'Accept: */*' -H  'User-Agent: python-requests/2.14.2' -H  'X-Auth-Token: 46a703f653f84629b5ac15cd145e69fc' 'http://127.0.0.1:9101/v1/executions/?limit=-11&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null'
{
    "faultstring": "Limit, \"-11\" specified, must be a positive number."
}%
```
and last but not the least, raise error with `-n 0` and suggest `-n -1` for full result set:

```
$ st2 execution list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/executions/?limit=0&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null
```

### CODE REVIEW

**Comment by armab** [comment]
good corner case find 👍

File: st2api/st2api/controllers/resource.py [hunk 1]
```diff
@@ -159,10 +159,19 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
...
-        if limit and int(limit) > self.max_limit:
-            # TODO: We should throw here, I don't like this.
-            msg = 'Limit "%s" specified, maximum value is "%s"' % (limit, self.max_limit)
-            raise ValueError(msg)
+        if limit:
+            # Display all the results
+            if int(limit) == -1:
+                limit = 0
+
+            if int(limit) <= -2:
```
Comment (Kami) [inline st2api/st2api/controllers/resource.py]: Yeah, good idea to handle that edge case here.

File: st2api/st2api/controllers/resource.py [hunk 2]
```diff
@@ -159,10 +159,19 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
         if offset >= 2**31:
             raise ValueError('Offset "%s" specified is more than 32-bit int' % (offset))
 
-        if limit and int(limit) > self.max_limit:
-            # TODO: We should throw here, I don't like this.
-            msg = 'Limit "%s" specified, maximum value is "%s"' % (limit, self.max_limit)
-            raise ValueError(msg)
+        if limit:
```
Comment (Kami) [inline st2api/st2api/controllers/resource.py]: Do we want to support `limit=0` and if yes, what should we do in such scenario? Does limit=0 even make sense?

Comment (humblearner) [reply st2api/st2api/controllers/resource.py]: Nope, but then default was always `limit=0`, now we are going `limit=-1` route. How should we block `limit=0` i.e ask user to use -1 or some positive value or throw some other message? And then with RBAC support we need to provide non-admin user a message if they use `limit=-1`

**Comment by Kami** [comment]
Also, what will we do with the default limit on the API side? IIRC, right now it's set to 100 for some API endpoints (notably executions), but not all.

File: st2api/st2api/controllers/resource.py (hunk 2 above)
Comment (Kami) [reply st2api/st2api/controllers/resource.py]: Yeah, maybe just throw / return an error message on limit=0 in query params.

**Comment by armab** [comment]
As I remember from the standup talk we agreed not to do anything RBAC-related for `-1` limit.

For max limit (currently 100), that looked like a consensus: https://github.com/StackStorm/st2/issues/3736#issuecomment-328860667
> set max_page_size to -1 (no limit) in st2.conf by default.
> This will optionally allow admins to adjust that st2.conf setting, but only if they really need it.

And of course, we leave `50` default limit (if no option provided) as it was already implemented before.

**Comment by Kami** [comment]
@armab 

> As I remember from the standup talk we agreed not to do anything RBAC-related for limit=-1.

We didn't fully agree on that yet. I was strongly for adding RBAC support for it.

And in case we don't do anything with RBAC, having `max_page_size` setting makes little sense because user can simply overwrite it with `limit=-1`.

**Comment by Kami** [comment]
Anything else left here?

Or we can go ahead and merge it?

**Comment by humblearner** [comment]
CLI help:
```bash
$ st2 trace list -h | grep -C 2 last
  -j, --json            Print output in JSON format.
  -y, --yaml            Print output in YAML format.
  -n LAST, --last LAST  List N most recent traces. Use -n -1 to fetch the full
                        result set.
  -s SORT_ORDER, --sort SORT_ORDER
```

**Comment by armab** [comment]
My only ask is to double check if it works for all CLI methods we have

**Comment by humblearner** [comment]
```bash
$ st2 execution list --last -1
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| id                       | action.ref | context.user | status                 | start_timestamp         | end_timestamp          |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| 5a50dbf1909a5052b4cdcbfd | core.local | stanley      | succeeded (1s elapsed) | Sat, 06 Jan 2018        | Sat, 06 Jan 2018       |
|                          |            |              |                        | 14:23:45 UTC            | 14:23:46 UTC           |
| 5a50dc15909a5052b4cdcc00 | core.local | stanley      | succeeded (0s elapsed) | Sat, 06 Jan 2018        | Sat, 06 Jan 2018       |
|                          |            |              |                        | 14:24:21 UTC            | 14:24:21 UTC           |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+

$ st2 execution list --last 1
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| id                       | action.ref | context.user | status                 | start_timestamp         | end_timestamp          |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
| 5a50dc15909a5052b4cdcc00 | core.local | stanley      | succeeded (0s elapsed) | Sat, 06 Jan 2018        | Sat, 06 Jan 2018       |
|                          |            |              |                        | 14:24:21 UTC            | 14:24:21 UTC           |
+--------------------------+------------+--------------+------------------------+-------------------------+------------------------+
+----------------------------------------------------------------------------------------------------------------------------------+
| Note: Only one action execution is displayed. Use -n/--last flag for more results.                                               |
+----------------------------------------------------------------------------------------------------------------------------------+

$ st2 execution list --last 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/executions/?limit=0&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null

$ st2 execution list --last -2
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-2" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/executions/?limit=-2&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null
```
```bash
$ st2 rule list -n 1
+----------------+---------+-------------------------------------------+---------+
| ref            | pack    | description                               | enabled |
+----------------+---------+-------------------------------------------+---------+
| chatops.notify | chatops | Notification rule to send results of      | True    |
|                |         | action executions to stream for chatops   |         |
+----------------+---------+-------------------------------------------+---------+
+--------------------------------------------------------------------------------+
| Note: Only one rule is displayed. Use -n/--last flag for more results.         |
+--------------------------------------------------------------------------------+

 $ st2 rule list -n -1
+-----------------------------------------+----------+-------------------------------------------+---------+
| ref                                     | pack     | description                               | enabled |
+-----------------------------------------+----------+-------------------------------------------+---------+
| chatops.notify                          | chatops  | Notification rule to send results of      | True    |
|                                         |          | action executions to stream for chatops   |         |
| examples.notify_inquiry                 | examples | Notify relevant users of an Inquiry       | False   |
|                                         |          | action                                    |         |
| examples.notify_slack                   | examples | Sample rule firing on action completion.  | True    |
| examples.sample_rule_file_watch         | examples | Sample rule custom trigger type - add a   | True    |
|                                         |          | file to be watched by file_watch_sensor   |         |
|                                         |          | in linux pack.                            |         |
| examples.sample_rule_with_actiontrigger | examples | Sample rule firing on action completion.  | True    |
| examples.sample_rule_with_timer         | examples | Sample rule using an Interval Timer.      | False   |
| examples.sample_rule_with_webhook       | examples | Sample rule dumping webhook payload to a  | True    |
|                                         |          | file.                                     |         |
| nagios.check_load                       | nagios   | Check cpu load on host                    | False   |
| nagios.check_load_procd                 | nagios   | Check D state procs on host when load is  | True    |
|                                         |          | high                                      |         |
| nagios.check_load_procs                 | nagios   | Check running procs on host when load is  | True    |
|                                         |          | high                                      |         |
| nagios.check_proc                       | nagios   | Check process state on host               | True    |
| nagios.notify_chat                      | nagios   | Post to chat when nagios service state    | False   |
|                                         |          | changes                                   |         |
| nagios.send_mail                        | nagios   | Send mail when nagios service state       | False   |
|                                         |          | changes                                   |         |
| twitter.relay_tweet_to_slack            | twitter  | Relay Tweet to the Slack channel.         | True    |
+-----------------------------------------+----------+-------------------------------------------+---------+

$ st2 rule list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/rules/?limit=0

$ st2 rule list -n -2
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-2" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/rules/?limit=-2
```
```bash
$ st2 trace list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/traces/?limit=0&sort_desc=True

$ st2 trace list -n -33
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-33" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/traces/?limit=-33&sort_desc=True

$ st2 trace list -n 1
id: 5a50de83909a505268b7818f
trace_tag: st2.IntervalTimer-2cc99c57-b5f9-4303-9c68-b00127e62398
start_timestamp: Sat, 06 Jan 2018 14:34:43 UTC
+--------------------------+------------------+-------------------------------------+-----------------------------+
| id                       | type             | ref                                 | updated_at                  |
+--------------------------+------------------+-------------------------------------+-----------------------------+
| 5a50de83909a505268b7818e | trigger-instance | core.2cc99c57-b5f9-4303-9c68-b00127 | 2018-01-06T14:34:43.292411Z |
|                          |                  | e62398                              |                             |
+--------------------------+------------------+-------------------------------------+-----------------------------+
+-----------------------------------------------------------------------------------------------------------------+
| Note: Only one trace is displayed. Use -n/--last flag for more results.                                         |
+-----------------------------------------------------------------------------------------------------------------+
```
```bash
$ st2 inquiry list -n 1
+--------------------------+-------+-------+------------+------+
| id                       | roles | users | route      | ttl  |
+--------------------------+-------+-------+------------+------+
| 5a50df47909a5052648ed3af |       |       | developers | 1440 |
+--------------------------+-------+-------+------------+------+
+---------------------------------------------------------------------------+
| Note: Only one inquiry is displayed. Use -n/--last flag for more results. |
+---------------------------------------------------------------------------+

$ st2 inquiry list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/exp/inquiries/?limit=0

$ st2 inquiry list -n -3
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-3" specified, must be a positive number. for url: http://127.0.0.1:9101/exp/inquiries/?limit=-3

$ st2 inquiry list -n 2
+--------------------------+-------+-------+------------+------+
| id                       | roles | users | route      | ttl  |
+--------------------------+-------+-------+------------+------+
| 5a50df52909a5052648ed3b3 |       |       | developers | 1440 |
| 5a50df47909a5052648ed3af |       |       | developers | 1440 |
+--------------------------+-------+-------+------------+------+
+----------------------------------------------------------------------------------+
| Note: Only first 2 inquiries are displayed. Use -n/--last flag for more results. |
+----------------------------------------------------------------------------------+
```
```bash
$ st2 rule-enforcement list
No matching items found

$ st2 rule-enforcement list -n 1
No matching items found

$ st2 rule-enforcement list -n -1
No matching items found

 $ st2 rule-enforcement list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/ruleenforcements/?limit=0

$ st2 rule-enforcement list -n -2
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-2" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/ruleenforcements/?limit=-2
```
```bash
$ st2 key list -n 1
+-------+---------+--------+-----------+--------------+------+------------------+
| name  | value   | secret | encrypted | scope        | user | expire_timestamp |
+-------+---------+--------+-----------+--------------+------+------------------+
| key_1 | value_1 | False  | False     | st2kv.system |      |                  |
+-------+---------+--------+-----------+--------------+------+------------------+
+----------------------------------------------------------------------------------+
| Note: Only one key value pair is displayed. Use -n/--last flag for more results. |
+----------------------------------------------------------------------------------+

$ st2 key list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/keys/?scope=system&decrypt=false&limit=0

$ st2 key list -n -1
+-------+---------+--------+-----------+--------------+------+------------------+
| name  | value   | secret | encrypted | scope        | user | expire_timestamp |
+-------+---------+--------+-----------+--------------+------+------------------+
| key_1 | value_1 | False  | False     | st2kv.system |      |                  |
| key_2 | value_2 | False  | False     | st2kv.system |      |                  |
+-------+---------+--------+-----------+--------------+------+------------------+

 
$ st2 key list -n -2
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-2" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/keys/?scope=system&decrypt=false&limit=-2


$ st2 trigger-instance list -n 1
+--------------------------+-------------------------------------+-------------------------------+-----------+
| id                       | trigger                             | occurrence_time               | status    |
+--------------------------+-------------------------------------+-------------------------------+-----------+
| 5a50ee19909a505268b7886a | core.2cc99c57-b5f9-4303-9c68-b00127 | Sat, 06 Jan 2018 15:41:13 UTC | processed |
|                          | e62398                              |                               |           |
+--------------------------+-------------------------------------+-------------------------------+-----------+
+------------------------------------------------------------------------------------------------------------+
| Note: Only one triggerinstance is displayed. Use -n/--last flag for more results.                          |
+------------------------------------------------------------------------------------------------------------+
```
```bash
$ st2 trigger-instance list -n 0
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "0" specified, must be a positive number or -1 for full result set. for url: http://127.0.0.1:9101/v1/triggerinstances/?limit=0

$ st2 trigger-instance list -n -3
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit, "-3" specified, must be a positive number. for url: http://127.0.0.1:9101/v1/triggerinstances/?limit=-3
```

File: st2client/st2client/commands/rule.py [hunk 3]
```diff
@@ -40,17 +40,21 @@ class RuleListCommand(resource.ResourceTableCommand):
...
+        self.default_limit = 50
+
         super(RuleListCommand, self).__init__(resource, 'list',
-                                              'Get the list of the 50 most recent %s.' %
-                                              resource.get_plural_display_name().lower(),
+                                              'Get the list of the %s most recent %s.' %
+                                              (self.default_limit,
+                                               resource.get_plural_display_name().lower()),
                                               *args, **kwargs)
         self.default_limit = 50
```
Comment (armab) [inline st2client/st2client/commands/rule.py]: Forgot to remove

Comment (humblearner) [reply st2client/st2client/commands/rule.py]: Damn! Thanks :)

**Comment by humblearner** [comment]
With this PR:

st2.conf limit stays intact, as @kami wanted, for general user:
```bash
$ st2 execution list -n 101
ERROR: 400 Client Error: Bad Request
MESSAGE: Limit "101" specified, maximum value is "100" for url: http://127.0.0.1:9101/v1/executions/?limit=101&exclude_attributes=result%2Ctrigger_instance&sort_desc=True&parent=null
```
and for admins, after RBAC implementation: 
```bash
$ st2 execution list -n -1 | grep stanley | wc -l
106
```
Users with proper privileges can access full result set. https://github.com/StackStorm/st2/pull/3706

File: st2api/st2api/controllers/v1/auth.py [hunk 4]
```diff
@@ -102,9 +102,18 @@ def get_all(self, requester_user, show_secrets=None, limit=None, offset=0):
         mask_secrets = self._get_mask_secrets(show_secrets=show_secrets,
                                               requester_user=requester_user)
 
-        if limit and int(limit) > self.max_limit:
-            msg = 'Limit "%s" specified, maximum value is "%s"' % (limit, self.max_limit)
-            raise ValueError(msg)
+        if limit:
```
Comment (Kami) [inline st2api/st2api/controllers/v1/auth.py]: It looks like this code is duplicated in two places and could perhaps be refactored in a utility function.

**Comment by Kami** [comment]
@humblearner A bit confused on that:

> and for admins, after RBAC implementation:

Unless I'm missing something, I don't see any RBAC checks in the code and any user (not just admins) can use `limit=-1`. Or am I missing something?

**Comment by armab** [comment]
From what I remember, per https://github.com/StackStorm/st2/issues/3736#issuecomment-328860667 as I understand we agreed to allow any user to utilize `-1` no limit by default with ability to enforce limits by server admin via st2.conf `max_page_size`.

I'm fine if similar limits or `-1` usage could be limited/enforced via additional non-default RBAC rules in case when st2 admins needs further control.

**Comment by Kami** [comment]
> From what I remember, per #3736 (comment) as I understand we agreed to allow any user to utilize -1 no limit by default with ability to enforce limits by server admin via st2.conf max_page_size.

> I'm fine if similar limits or -1 usage could be limited/enforced via additional non-default RBAC rules in case when st2 admins needs further control.

Yeah, I remember we had troubles reaching an agreement since I was against allowing it for everyone :)

Without limiting `-1` to admins only and / or adding additional "limit_all" permission, `max_paze_size` in config doesn't provide much value.

One middle of the ground approach would be to add a config flag, e.g. "require_admin_for_list_all = True" and default it to False (although, I still think it would be better and safer to default it to True, but it seems not all others agree with me).

In any case, this is not a blocker for that PR, but I would like to get a consensus on that.

File: st2api/st2api/controllers/v1/auth.py (hunk 4 above)
Comment (Kami) [reply st2api/st2api/controllers/v1/auth.py]: That's my only "blocker", besides that, LGTM.

I would also like to reach some kind of consensus on that - https://github.com/StackStorm/st2/pull/3761#issuecomment-356313645

**Pull request merged**